desktop/cmp/grid/find/GridFindFieldModel.js

```diff
@@ -0,0 +1,324 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2021 Extremely Heavy Industries Inc.
+ */
+import {HoistModel, XH} from '@xh/hoist/core';
+import {FieldType} from '@xh/hoist/data';
+import {action, bindable, computed, observable, makeObservable} from '@xh/hoist/mobx';
+import {stripTags, throwIf} from '@xh/hoist/utils/js';
+import {
+    escapeRegExp,
+    filter,
+    flatMap,
+    get,
+    intersection,
+    isNil,
+    isArray,
+    isEmpty,
+    isEqual,
+    isFinite,
+    without
+} from 'lodash';
+
+export class GridFindFieldModel extends HoistModel {
+
+    /** @member {GridModel} */
+    gridModel;
+    /** @member {string} */
+    matchMode;
+    /** @member {number} */
+    queryBuffer;
+    /** @member {string[]} */
+    includeFields;
+    /** @member {string[]} */
+    excludeFields;
+
+    @bindable query;
+    @observable.ref results;
+    @observable.ref records;
+
+    get count() {
+        return this.results?.length;
+    }
+
+    get selectedIdx() {
+        if (!this.count) return null;
+        const matchIdx = this.results.indexOf(this.gridModel.selectedId);
+        return matchIdx > -1 ? matchIdx : null;
+    }
+
+    get countLabel() {
+        if (isNil(this.results)) return null;
+        const {count, selectedIdx} = this,
+            match = isFinite(selectedIdx) ? selectedIdx + 1 : 0;
+        return `${match}/${count}`;
+    }
+
+    @computed
+    get hasQuery() {
+        return !isNil(this.query) && this.query.length > 0;
+    }
+
+    @computed
+    get hasResults() {
+        return !isNil(this.results) && !isEmpty(this.results);
+    }
+
+    constructor({
+        gridModel,
+        matchMode = 'startWord',
+        queryBuffer = 200,
+        includeFields,
+        excludeFields
+    }) {
+        super();
+        makeObservable(this);
+
+        this.gridModel = gridModel;
+        this.matchMode = matchMode;
+        this.queryBuffer = queryBuffer;
+        this.includeFields = includeFields;
+        this.excludeFields = excludeFields;
+
+        throwIf(!gridModel, "Must specify 'gridModel' in GridFindField.");
+
+        this.addReaction({
+            track: () => [
+                this.hasQuery,
```

Comment:
I know that `hasQuery` came in at my suggestion around readying this model's view on records in a lazy manner, but I think these reactions could be optimized a bit further. Maybe I am making too much out of this, but:

- We don't need to rebuild the records in the (common) case where the user enters a query, blanks it out, then enters another query.
- These two reactions call `updateResults` twice whenever the query goes from nil -> something.

Feels like there might be a clearer way to handle - potentially by having a lazily cached field/getter for the processed records that is cleared out whenever the store data changes and regenerated again whenever it's asked for by a call to updateResults() with a non-empty query.